### PR TITLE
[6888] Carousel section

### PR DESCRIPTION
**Pull request by SimonDEvans**
Closes #6888

**Link to sample test page**:
Locally: http://localhost:8000/youtube-regrets-2021/
Omitted prev/next buttons on mobile

**Screencap**

https://user-images.githubusercontent.com/2553896/124470045-004e6200-dd93-11eb-96f0-90e2e925bb65.mov

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{% load static i18n %}
+
+<div class="carousel" data-carousel data-slidetotal="5" id="yt-regrets-carousel">
+  <div class="carousel__controls">
+    <div class="carousel__controls-inner">
+      <div class="d-flex align-items-center justify-content-end" data-glide-el="controls">
+        <button class="carousel__button carousel__button--prev button mr-3" aria-label="Previous slide" data-glide-dir="<">
+          <svg width="41" height="41" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20.48 39.736c10.586 0 19.167-8.581 19.167-19.167 0-10.585-8.58-19.166-19.166-19.166S1.314 9.983 1.314 20.569s8.581 19.167 19.167 19.167z" stroke="#000" stroke-linecap="round" stroke-linejoin="round"/><path class="arrow" d="M20.48 12.903l-7.666 7.666 7.667 7.667M28.147 20.57H12.814" stroke="#000" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (helenb) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html:8]: can you add `aria-hidden="true"` to the svgs

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  <div data-glide-el="track" class="carousel__container glide__track">
+    <ul class="carousel__slides glide__slides" aria-live="polite">
+      <li class="carousel__item glide__slide">
+        <div class="carousel__image">
+          <img src="{% static "_images/youtube-regrets/carousel/quote-1.png" %}" alt="">
+        </div>
+        <div class="carousel__item-content">
+          <img class="carousel__speech-bubble-desktop mb-4" src="{% static "_images/youtube-regrets/speech-bubble-sad.svg" %}" alt="">
+          <p class="carousel__item-text">
+            &ldquo;{% trans "In coming out to myself and close friends as transgender, <span>my biggest regret was turning to YouTube to hear the stories of other trans and queer people.</span> Simply typing in the word 'transgender' brought up countless videos that were essentially describing my struggle as a mental illness and as something that shouldn't exist. YouTube reminded me why I hid in the closet for so many years." %}
```
Comment (helenb) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html:25]: I believe if you have tags inside a `trans` tag you need to use `blocktrans` instead

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          </p>
+          <p class="carousel__item-text">
+            {% trans "Every now and then YouTube will continue to recommend me a video that tells me that my gender identity is wrong — and it reminds me of how much hate is squarely directed at me and people like me. I'm somewhat older, I've been dealing with these issues internally for a long time and I have therapy to work out these issues, but I can't imagine what it's like for those without access to help. <span>YouTube is a part of my pain in coming out</span> and is a reminder of how terrible this world can be to those who are different. I have to be proud in spite of places like YouTube." %}&rdquo;
```
Comment (helenb) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html:28]: Same point re `blocktrans`

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        </div>
+      </li>
+      <li class="carousel__item glide__slide">
+        <div class="carousel__image">
+          <img src="{% static "_images/youtube-regrets/carousel/quote-2.png" %}" alt="">
+        </div>
+        <div class="carousel__item-content">
+          <img class="carousel__speech-bubble-desktop mb-4" src="{% static "_images/youtube-regrets/speech-bubble-sad.svg" %}" alt="">
+          <p class="carousel__item-text">
+            &ldquo;{% trans "When my son was preschool age, he liked to watch 'Thomas the Tank Engine' videos on YouTube. One time when I checked on him, he was watching a video compilation that <span>contained graphic depictions of train wrecks.</span>" %}&rdquo;
```
Comment (helenb) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html:39]: same point re blocktrans

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        </div>
+      </li>
+      <li class="carousel__item glide__slide">
+        <div class="carousel__image">
+          <img src="{% static "_images/youtube-regrets/carousel/quote-3.png" %}" alt="">
+        </div>
+        <div class="carousel__item-content">
+          <img class="carousel__speech-bubble-desktop mb-4" src="{% static "_images/youtube-regrets/speech-bubble-sad.svg" %}" alt="">
+          <p class="carousel__item-text">
+            &ldquo;{% trans "My eight-year-old granddaughter was using our tablet to watch children's videos; unknowingly as grandparents we thought she was safe. The next day we watched as our granddaughter sat at the dinner table <span>speaking candidly to her parents about her fear of cancer.</span>" %}
```
Comment (helenb) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/carousel.html:50]: Same point re blocktrans... I'll stop repeating myself now, I'm sure you can find them all :)

**Review by helenb — COMMENTED** [review]
Hi @SimonDEvans, all looks good - just a few minor points. On testing I noticed that you didn't have the carousel arrows any more - they are in the mobile designs at the bottom of the slide. Only other query is whether the css would be better in the main `youtube-regrets-2021.scss as Chris had put all the other css in there - but I guess there is enough new code to justify the separate file.

**Comment by SimonDEvans** [comment]
> Hi @SimonDEvans, all looks good - just a few minor points. On testing I noticed that you didn't have the carousel arrows any more - they are in the mobile designs at the bottom of the slide. Only other query is whether the css would be better in the main `youtube-regrets-2021.scss as Chris had put all the other css in there - but I guess there is enough new code to justify the separate file.

@helenb thanks for the review! Translation stuff should be sorted now.

Re carousel buttons: yeah I've left them out and documented that [here.](https://github.com/mozilla/foundation.mozilla.org/issues/6888#issuecomment-874068104) It would be nice to add them though.

**Comment by helenb** [comment]
Cool, @chris-lawton this can be merged to the integration branch now.

**Pull request merged**